sphinx/directives/other.py

```diff
@@ -361,6 +364,92 @@ def run(self) -> List[Node]:
         return super().run()
 
 
+class IncludeDocument(SphinxDirective):
+
+    """ Include one document in another. The language need not match  """
+
+    required_arguments = 1
+    optional_arguments = 0
+    final_argument_whitespace = True
+    option_spec = {'start-line': int,
+                   'end-line': int,
+                   'start-after': directives.unchanged_required,
+                   'end-before': directives.unchanged_required,
+                   'encoding': directives.encoding}
+
+    standard_include_path = BaseInclude.standard_include_path
+
+    def run(self):
+        """Include a file as part of the content of this reST file."""
+
+        # copied from docutils.parsers.rst.directives.misc.Include
+        if not self.state.document.settings.file_insertion_enabled:
+            raise self.warning('"%s" directive disabled.' % self.name)
+        source = self.state_machine.input_lines.source(
+            self.lineno - self.state_machine.input_offset - 1)
+        source_dir = os.path.dirname(os.path.abspath(source))
+        path = directives.path(self.arguments[0])
+        if path.startswith('<') and path.endswith('>'):
+            path = os.path.join(self.standard_include_path, path[1:-1])
+        path = os.path.normpath(os.path.join(source_dir, path))
+        path = relative_path(None, path)
+        path = nodes.reprunicode(path)
+        encoding = self.options.get(
+            'encoding', self.state.document.settings.input_encoding)
```

Comment:
`config.source_encoding` is better for the default value.

Reply:
This code is copied from the built-in `..include`